http/rest-client/src/test/java/io/quarkus/ts/http/restclient/GzipMaxInputIT.java

```diff
@@ -0,0 +1,127 @@
+package io.quarkus.ts.http.restclient;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import jakarta.ws.rs.core.MediaType;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class GzipMaxInputIT {
+
+    final byte[] zero_bytes = new byte[0];
+
+    final String invalid_value = "";
+    final byte[] gzip_bellow_1K_limit_max = new byte[900];
+    final byte[] gzip_same_1K_limit_max = new byte[1024];
+    final byte[] gzip_over_1K_limit_max = new byte[10000000];
+
+    @QuarkusApplication(classes = { GzipResource.class, GzipClientService.class }, properties = "gzip.properties")
+    static RestService app = new RestService();
+
+    /**
+     *
+     * According "All configurations options" guide the property 'quarkus.resteasy.gzip.max-input' refers to
+     *
+     * Maximum deflated file bytes size
+     *
+     * If the limit is exceeded, Resteasy will return Response with status 413("Request Entity Too Large")
+     *
+     */
+
+    @Test
+    void invalidValue() throws IOException {
+        app.given()
+                .contentType(MediaType.TEXT_PLAIN)
+                .header("Content-Encoding", "gzip")
+                .body(invalid_value)
+                .when()
+                .post("/gzip")
+                .then()
+                .statusCode(400);
+    }
+
+    @Disabled("Because https://github.com/quarkusio/quarkus/issues/40806")
+    @Test
+    void sendZeroValue() throws IOException {
+        byte[] compressedData = generateCompressedData(zero_bytes);
+
+        Response response = app.given()
```

Comment:
The scenario seems too complicated, either you go this way and you don't need restcliet at all.
Or you define rest client and inject it into the test case as in https://github.com/quarkusio/quarkus/issues/39636 reproducer. But that work just in case of `@QuarkusTest` where the same JVM is used.

I personally like the "rest client" approach similar to this:
```java
    @Test
    void failAlways() {
        WebApplicationException ex = Assertions.assertThrows(WebApplicationException.class, () -> client.gzip(new byte[20*1024*1024]),
                "This test is not supposed to ever fail, 20M is more than defined quarkus.resteasy.gzip.max-input=100k and RESTEasy 10M default");
        assertEquals(HttpStatus.SC_REQUEST_TOO_LONG, ex.getResponse().getStatus());
    }

    @Test
    void failOn2M() {
        WebApplicationException ex = Assertions.assertThrows(WebApplicationException.class, () -> client.gzip(new byte[2*1024*1024]),
                "Fails when quarkus.resteasy.gzip.max-input=100k is not parsed and RESTEasy falls back to 10M default");
        assertEquals(HttpStatus.SC_REQUEST_TOO_LONG, ex.getResponse().getStatus());
    }

    @Test
    void passOn2k() {
        client.gzip(new byte[2*1024]);
    }

```
Instead of calling `client.gzip(...)` you would call helper method that would call 
```
app.given()
                .contentType(MediaType.APPLICATION_OCTET_STREAM)
                .header("Content-Encoding", "gzip")
                .body(compressedData)
                .when()
                .post("/gzip")
                .then()
                .extract().response();
```